src/Data/List/NonEmpty/Properties.agda

```diff
@@ -49,6 +58,44 @@ toList->>= : ∀ {ℓ} {A B : Set ℓ}
              (List⁺.toList (xs >>= f))
 toList->>= f (x ∷ xs) = begin
   List.concat (List.map (List⁺.toList ∘ f) (x ∷ xs))
-    ≡⟨ cong List.concat $ map-compose {g = List⁺.toList} (x ∷ xs) ⟩
+    ≡⟨ cong List.concat $ Listₚ.map-compose {g = List⁺.toList} (x ∷ xs) ⟩
   List.concat (List.map List⁺.toList (List.map f (x ∷ xs)))
     ∎
+
+length-++⁺ : (xs : List A) (ys : List⁺ A) → length (xs ++⁺ ys) ≡ List.length xs + length ys
+length-++⁺ [] ys                                = refl
+length-++⁺ (x ∷ xs) ys rewrite length-++⁺ xs ys = refl
+
+length-++⁺-tail : (xs : List A) (ys : List⁺ A) → length (xs ++⁺ ys) ≡ suc (List.length xs + List.length (List⁺.tail ys))
+length-++⁺-tail [] ys                                     = refl
+length-++⁺-tail (x ∷ xs) ys rewrite length-++⁺-tail xs ys = refl
+
+++-++⁺ : (xs : List A) → ∀ {ys zs} → (xs ++ ys) ++⁺ zs ≡ xs ++⁺ ys ++⁺ zs
+++-++⁺ []      = refl
+++-++⁺ (x ∷ l) = cong (x ∷_) (cong toList (++-++⁺ l))
+
+++⁺-cancelˡ′ : ∀ xs ys {zs ws : List⁺ A} → xs ++⁺ zs ≡ ys ++⁺ ws → List.length xs ≡ List.length ys → zs ≡ ws
```

Comment:
Again can we introduce the variables in alphabetical order (same for the lemma below)?

Reply:
adjusted